src/test/java/com/shapesecurity/shift/es2016/test262/Test262.java

```diff
@@ -0,0 +1,385 @@
+package com.shapesecurity.shift.es2016.test262;
+
+import com.shapesecurity.functional.data.ImmutableList;
+import com.shapesecurity.shift.es2016.ast.Program;
+import com.shapesecurity.shift.es2016.ast.Script;
+import com.shapesecurity.shift.es2016.parser.EarlyError;
+import com.shapesecurity.shift.es2016.parser.EarlyErrorChecker;
+import com.shapesecurity.shift.es2016.parser.JsError;
+import com.shapesecurity.shift.es2016.parser.Parser;
+import org.junit.Assert;
+import org.junit.Test;
+import org.yaml.snakeyaml.Yaml;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class Test262 {
+
+	private static final Yaml yamlParser = new Yaml();
+
+	private static final HashSet<String> xfailParse = new HashSet<>(Arrays.asList(
+			// shift-java#198
+			"src/test/resources/test262/test/annexB/language/statements/for-in/nonstrict-initializer.js",
+			"src/test/resources/test262/test/annexB/language/statements/for-in/bare-initializer.js",
+			"src/test/resources/test262/test/annexB/language/statements/for-in/strict-initializer.js",
+			"src/test/resources/test262/test/annexB/language/statements/for-in/var-objectbindingpattern-initializer.js",
+			"src/test/resources/test262/test/annexB/language/statements/for-in/var-arraybindingpattern-initializer.js",
+
+			// async/await not implemented
```

Comment:
Can't we filter these based on feature flag? We should whitelist/blacklist supported/unsupported feature flags.

Reply:
No `async` feature flag exists, the flag is much too broad. Trailing commas, etc, do not have flags or feature flags to accompany them.

Reply:
@Protryon There are `async-functions` and `async-iteration` feature flags which cover most of them (and the exceptions are arguably bugs in the test suite). That's [what shift-parser-js does](https://github.com/shapesecurity/shift-parser-js/blob/f973b067adbb7e72fe3a010596bc50c73a5e6f3c/test/test262/expectations/index.js#L13-L14).

I'm surprised this looks so different from the shift-parser-js tests, in fact. For example, there's [well over a hundred trailing comma tests](https://github.com/shapesecurity/shift-parser-js/blob/f973b067adbb7e72fe3a010596bc50c73a5e6f3c/test/test262/expectations/trailing-function-commas.js#L174), but I only see reference to 14 here.

Reply:
@bakkot `async-functions` covers 4 tests, and `async-iteration` covers none (doesn't exist), at least in the commit currently in the repo. As for all the trailing comma tests, they are running and passing. I suppose that in itself could be a bug in the parser. There is no feature flag for trailing commas, however a hack would be to check if the name contains `trailing-comma`.

Reply:
@Protryon ah, that's because the test262 you're using is two years out of date. It should probably be updated - try ` 72f1cfa`.

(Yes, that will mean pulling in tests for new features, but that's OK. And it matches what shift-parser-js does.)

Reply:
> There is no feature flag for trailing commas, however a hack would be to check if the name contains trailing-comma.

Just list them out manually, [like shift-parser-js](https://github.com/shapesecurity/shift-parser-js/blob/f973b067adbb7e72fe3a010596bc50c73a5e6f3c/test/test262/expectations/trailing-function-commas.js). I would expect this project's test expectations to look almost exactly like shift-parser-js's.